nucypher/network/middleware.py

```diff
@@ -36,33 +36,53 @@ class NotFound(UnexpectedResponse):
     pass
 
 
-class RestMiddleware:
-    log = Logger()
-
-    class _Client:
+class NucypherMiddlewareClient:
+    library = requests
+
+    def parse_node_or_host_and_port(self, node, host, port):
+        raise NotImplementedError()
+
+    def invoke_method(self, method, url, *args, **kwargs):
+        raise NotImplementedError()
+
+    def __getattr__(self, method_name):
+        # Quick sanity check.
+        if not method_name in ("post", "get", "put", "patch", "delete"):
+            raise TypeError(
+                "This client is for HTTP only - you need to use a real HTTP verb, not '{}'.".format(method_name))
+
+        def method_wrapper(path, node=None, host=None, port=None, *args, **kwargs):
+            host, certificate_filepath, http_client = self.parse_node_or_host_and_port(node, host, port)
+            method = getattr(http_client, method_name)
+
+            url = "https://{}/{}".format(host, path)
+            response = self.invoke_method(method, url, verify=certificate_filepath, *args, **kwargs)
+            cleaned_response = self.response_cleaner(response)
+            if cleaned_response.status_code >= 300:
+                if cleaned_response.status_code == 404:
+                    raise NotFound(
+                        "While trying to {} {} ({}), server claims not to have found it.  Response: {}".format(
```

Comment:
Great candidate for a f-string.

Reply:
Come on, @tuxxy, don't be so {} on this.  If you can't read "While trying to {} {} ({})", you're basically {}.

Reply:
What is this? Like a Python madlib?

Reply:
f"One day, my {noun} got stuck inside a {noun}.  And you won't believe who came along to {verb} it for me: {proper_noun}!  What an {adjective} experience!"